tests/unit_tests/hyperion/external_interaction/test_agamemnon.py

```
@@ -0,0 +1,184 @@
+    mock_logger: MagicMock,
+    loop_name: str,
+):
+    assert (
+        _get_pin_type_from_agamemnon_parameters(params_from_loop_type(loop_name))
+        == SinglePin()
+    )
+    mock_logger.warning.assert_called_once()
+
+
+@pytest.mark.parametrize(
+    "loop_name",
+    [
+        "multipin_67x56",
+        "multipin_90+4",
+        "multipin_8",
+        "multipin_6x50+",
+        "multipin_6x50+98.",
+        "multipin_6x50+.1",
+        "multipin_6x.50+98",
+        "multipin_6x50+98.1.2",
+        "multipin_6x50.5.6+98",
+        "multipin_6x50+98..1",
+        "multipin_6x.50+.98",
+        "multipin_6x+98",
+    ],
+)
+def test_given_unrecognised_multipin_in_parameters_then_warning_logged_single_pin_returned(
+    loop_name: str,
+):
+    with pytest.raises(ValueError) as e:
+        _get_pin_type_from_agamemnon_parameters(params_from_loop_type(loop_name))
+    assert "Expected multipin format" in str(e.value)
+
+
+def configure_mock_agamemnon(mock_requests: MagicMock, loop_type: str | None):
+    mock_requests.get.return_value.content = json.dumps(
+        {"collect": params_from_loop_type(loop_type)}
+    )
+
+
+@patch("mx_bluesky.hyperion.external_interaction.agamemnon.requests")
+def test_when_get_next_instruction_called_then_expected_agamemnon_url_queried(
+    mock_requests: MagicMock,
+):
+    configure_mock_agamemnon(mock_requests, None)
+    get_next_instruction("i03")
+    mock_requests.get.assert_called_once_with(
+        "http://agamemnon.diamond.ac.uk/getnextcollect/i03",
+        headers={"Accept": "application/json"},
+    )
+
+
+@patch("mx_bluesky.hyperion.external_interaction.agamemnon.requests")
+def test_given_agamemnon_returns_an_unexpected_response_then_exception_is_thrown(
+    mock_requests: MagicMock,
+):
+    mock_requests.get.return_value.content = json.dumps({"not_collect": ""})
+    with pytest.raises(Exception) as e:
```

Nit: Following my last nit, even though we swallow these errors, raising `KeyError` would let use do:
```suggestion
    with pytest.raises(KeyError) as e:
```
which I generally like more in unit tests for specificity.